src/components/DynamicComponent/index.js

```diff
@@ -36,31 +36,38 @@ class DynamicComponent extends Component {
     this.state = {}
   }
   componentDidMount() {
-    this.props
-      .require(this.props.src)
-      .then(module => {
-        this.setState({
-          LoadedComponent: module.hasOwnProperty('default')
-            ? module['default']
-            : module
+    if (this.props.src) {
+      this.props
+        .require(this.props.src)
+        .then(module => {
+          this.setState({
+            LoadedComponent: module.hasOwnProperty('default')
+              ? module['default']
+              : module
+          })
         })
-      })
-      .catch(error => {
-        this.setState({ error })
-      })
+        .catch(error => {
+          this.setState({ error })
+        })
+    }
   }
   render() {
     const { LoadedComponent } = this.state
-
+    const IdentifierComponent = this.props.identifiers[this.props.identifier]
     const error =
-      this.state.error || (!this.props.src && new Error('Missing src'))
+      this.state.error ||
+      (!this.props.src &&
+        !IdentifierComponent &&
+        new Error('Missing src or identifier'))
     if (error) {
       throw error
     }
 
-    if (LoadedComponent) {
+    const DisplayComponent = LoadedComponent || IdentifierComponent
+
+    if (DisplayComponent) {
       return (
-        <LoadedComponent require={this.props.require} {...this.props.props} />
+        <DisplayComponent require={this.props.require} {...this.props.props} />
```

Comment:
To make out life easier for placeholders and such: let's always add the `identifier` as prop:
```suggestion
        <DisplayComponent require={this.props.require} identifier={this.props.identifier} {...this.props.props} />
```